instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_1;
+
+import static io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest.controller;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.CAPTURE_HEADERS;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.ERROR;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.EXCEPTION;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.INDEXED_CHILD;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.PATH_PARAM;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.QUERY_PARAM;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.REDIRECT;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.SUCCESS;
+
+import org.restlet.Context;
+import org.restlet.data.Form;
+import org.restlet.data.MediaType;
+import org.restlet.data.Reference;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+import org.restlet.resource.Representation;
+import org.restlet.resource.Resource;
+import org.restlet.resource.StringRepresentation;
+import org.restlet.resource.Variant;
+
+public class RestletAppTestBase {
+  abstract static class BaseResource extends Resource {
+
+    @Override
+    public void init(Context context, Request request, Response response) {
+      super.init(context, request, response);
+      getVariants().add(new Variant(MediaType.TEXT_PLAIN));
+    }
+  }
+
+  public static class SuccessResource extends BaseResource {
+
+    @Override
+    public Representation represent(Variant variant) {
+      return controller(SUCCESS, () -> new StringRepresentation(SUCCESS.getBody()));
+    }
+  }
+
+  public static class ErrorResource extends BaseResource {
+
+    @Override
+    public Representation represent(Variant variant) {
+      return controller(
+          ERROR,
+          () -> {
+            getResponse().setStatus(Status.valueOf(ERROR.getStatus()), ERROR.getBody());
+            return new StringRepresentation(ERROR.getBody());
+          });
+    }
+  }
+
+  public static class ExceptionResource extends BaseResource {
+
+    @Override
+    public Representation represent(Variant variant) {
+      return controller(
+          EXCEPTION,
+          () -> {
+            throw new IllegalStateException(EXCEPTION.getBody());
+          });
+    }
+  }
+
+  public static class QueryParamResource extends BaseResource {
```

Comment:
```suggestion
  static class QueryParamResource extends BaseResource {
```

Reply:
these need to be public so jetty can access the constructors. When testing them as package private i see the following:


> 18:26:18.461 [qtp1342212468-54] WARN  org.eclipse.jetty.util.log - Exception while instantiating the target handler.
java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.restlet.v1_1.RestletAppTestBase$QueryParamResource.<init>()


and for `PathParamResource`


> 18:27:52.806 [qtp111210112-54] WARN  org.eclipse.jetty.util.log - Exception while instantiating the target handler.
java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.restlet.v1_1.RestletAppTestBase$PathParamResource.<init>()

etc